tools/functions_missing_types.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+"""Find the functions in a module missing type annotations.
+
+To use it run
+
+./functions_missing_types.py <module>
+
+and it will print out a list of functions in the module that don't
+have types.
+
+"""
+import argparse
+import ast
+import importlib
+import os
+
+NUMPY_ROOT = os.path.dirname(os.path.join(
+    os.path.abspath(__file__), "..",
+))
+
+# Technically "public" functions (they don't start with an underscore)
+# that we don't want to include.
+EXCLUDE_LIST = {
+    "numpy": {
+        # Stdlib modules in the namespace by accident
+        "absolute_import",
+        "division",
+        "print_function",
+        "warnings",
+        # Accidentally public, deprecated, or shouldn't be used
+        "Tester",
+        "add_docstring",
+        "add_newdoc",
+        "add_newdoc_ufunc",
+        "core",
+        "fastCopyAndTranspose",
+        "get_array_wrap",
+        "int_asbuffer",
+        "oldnumeric",
+        "safe_eval",
+        "set_numeric_ops",
+        "test",
+        # Builtins
+        "bool",
+        "complex",
+        "float",
+        "int",
+        "long",
+        "object",
+        "str",
+        "unicode",
+        # Should use numpy_financial instead
+        "fv",
+        "ipmt",
+        "irr",
+        "mirr",
+        "nper",
+        "npv",
+        "pmt",
+        "ppmt",
+        "pv",
+        "rate",
+        # More standard names should be preferred
+        "alltrue",  # all
+        "sometrue",  # any
```

Comment:
Could you add `alen` to the list?
It's been deprecated as of https://github.com/numpy/numpy/pull/14181.

```suggestion
        "sometrue",  # any
        "alen",
```

Reply:
Done in https://github.com/numpy/numpy/pull/16629/commits/2d90622bbf90213d75b1064a40b28e6b20c5d212.